src/multibody/fcl.hpp

```diff
@@ -84,38 +86,57 @@ struct GeometryObject
 
   /// \brief Index of the parent frame
   ///
-  /// Parent frame may be unset (set to -1) as it is mostly used as a documentation of the tree, or in third-party libraries.
+  /// Parent frame may be unset (to the std::numeric_limits<FrameIndex>::max() value) as it is mostly used as a documentation of the tree, or in third-party libraries.
   /// The URDF parser of Pinocchio is setting it to the proper value according to the urdf link-joint tree.
   /// In particular, anchor joints of URDF would cause parent frame to be different to joint frame.
   FrameIndex parentFrame;
 
   /// \brief Index of the parent joint
   JointIndex parentJoint;
 
-  /// \brief The actual cloud of points representing the collision mesh of the object after scaling.
-  CollisionGeometryPtr fcl;
+  /// \brief The FCL CollisionGeometry (might be a Mesh, a Geometry Primitive, etc.)
+  CollisionGeometryPtr geometry;
+
+  /// \brief The former pointer to the FCL CollisionGeometry.
+  /// \deprecated It is now deprecated and has been renamed GeometryObject::geometry
+  PINOCCHIO_DEPRECATED CollisionGeometryPtr & fcl;
```

Comment:
After some trial and error, this could be replaced by the safer version below.

```cpp
union {
  CollisionGeometryPtr geometry;
  PINOCCHIO_DEPRECATED CollisionGeometryPtr fcl;
}
```

I didn't check how doxygen behaves.

Reply:
I've tried, and I got the following error:
```
union member 'geometry' has a non-trivial copy constructor
because field of type 'boost::detail::shared_count' has a user-provided copy constructor
```

Reply:
Does the solution I've suggested work?

Reply:
My bad. I actually tested it only with doubles...

Reply:
I will give a try to HPP with the devel branch of pinocchio in the next few days.

Reply:
I launched a compilation with these changes and the one in https://github.com/humanoid-path-planner/hpp-fcl/pull/119.

Reply:
Sounds great. After your feedback, I will make a new release of hpp-fcl to sync anaconda.

Reply:
It worked fine.